fuzz/Cargo.toml

```diff
@@ -0,0 +1,40 @@
+
+[package]
+name = "quinn-fuzz"
+version = "0.0.0"
+authors = ["Automatically generated"]
+publish = false
+edition = "2018"
+
+[package.metadata]
+cargo-fuzz = true
+
+[dependencies]
+arbitrary = { version = "0.4.5", features = ["derive"] }
+libfuzzer-sys = "0.3"
+bytes = "0.5.6"
+
+[dependencies.quinn]
+path = "../quinn"
+
+[dependencies.proto]
+features = ["unstructured-data"]
+path = "../quinn-proto"
+package = "quinn-proto"
+version = "0.6.1"
+
+# Prevent this from interfering with workspaces
+[workspace]
+members = ["."]
```

Comment:
Why not make this a first-class member of the workspace?

Reply:
This turned out to be quite the unexpected challenge! 

When trying to do `cargo build ---all-workspaces` the build fails from an unresolved import error:
```rust
error[E0432]: unresolved import `proto::fuzzing`
 --> fuzz/fuzz_targets/streams.rs:7:12
  |
7 | use proto::fuzzing::{ResetStream, Streams};
  |            ^^^^^^^ could not find `fuzzing` in `proto`
error: could not compile `quinn-fuzz`.
```

We re-export these in `quinn-proto` behind a `cfg(fuzzing)` so it isn't surprising to see that error. 
So then the problem seemed to be: how to set a `cfg` in a workspace. I have not found a solution for it!

I've tried:
- adding a build script to add the necessary rustc flags, e.g. `println!("cargo:rustc-cfg=fuzzing");` 
- addding a .cargo/config.toml file with rustflags config
- settings features block in fuzz/Cargo.toml 

I suppose another approach would be to change the commands in the github action, but I don't have a clear idea yet how that'd work. 

When building and running the fuzzer I am using tools from [cargo-fuzz](https://github.com/rust-fuzz/cargo-fuzz), and they pass these flags seen here: https://github.com/rust-fuzz/cargo-fuzz/blob/4b7a6a674b3b6f7509c9568462b10cde5fcf841c/src/project.rs#L159
```rust
        let mut rustflags: String = "--cfg fuzzing \
                                     -Cpasses=sancov \
                                     -Cllvm-args=-sanitizer-coverage-level=4 \
                                     -Cllvm-args=-sanitizer-coverage-trace-compares \
                                     -Cllvm-args=-sanitizer-coverage-inline-8bit-counters \
                                     -Cllvm-args=-sanitizer-coverage-pc-table \
                                     -Clink-dead-code"
```

even adding these verbatim to build script did not change the failure of `cargo build --all-workspaces`. 
I would like to get this to build properly and will continue to ask around for ideas. 
In the meantime I will move fuzzing out of the primary workspace and get the CI back upright.

Reply:
In that case I think it's fine for this to not be a workspace member; I clearly underestimated the significance of the barrier here!